File: dart/9.dart
class Solution {
  bool isPalindrome(int x) {
    final xStr = x.toString();
    final bound = xStr.length ~/ 2;

    for (int i = 0; i < bound; i++) {
      if (xStr[i] != xStr[xStr.length - i - 1]) return false;
      continue;
    }
    return true;
  }
}

File: rust/src/codes/9.rs
struct Solution {}
impl Solution {
    pub fn is_palindrome(x: i32) -> bool {
        let x_str = x.to_string();
        let bytes = x_str.as_bytes();
        let len = bytes.len();

        if len <= 1 {
            return true;
        }

        for i in 0..len / 2 {
            if bytes[i] != bytes[len - i - 1] {
                return false;
            }
        }
        true
    }
}
